src/main/java/Duke.java

```diff
@@ -1,10 +1,186 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
+
+    static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void writeTaskToFile(String filePath) {
+        try {
+            FileWriter fw = new FileWriter(filePath);
+            for (Task taskToAdd : tasks) {
+                fw.write(taskToAdd.writeToFile() + "\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    public static void readTask(String filePath) {
+        try {
+            File taskList = new File(filePath);
+            Scanner sc = new Scanner(taskList);
+            while (sc.hasNextLine()) {
+                tasks.add(convertTextToTask(sc.nextLine()));
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    private static Task convertTextToTask(String line) {
+        String[] taskInfo = line.split("\\|");
+        Task convertedTask;
+
+        if (taskInfo[0].equals("T")) {
+            convertedTask = new ToDo(taskInfo[2]);
+        } else if (taskInfo[0].equals("E")) {
+            convertedTask = new Event(taskInfo[2], taskInfo[3]);
+        } else {
+            convertedTask = new Deadline(taskInfo[2], taskInfo[3]);
+        }
+
+        if (taskInfo[1].equals("1")) {
+            convertedTask.markDone();
+        }
```

Comment:
Consider using an enum for your string comparisons.
The reason is, if in the future, you want to change the character label, it will be easier and clearer.

You may also consider having a temporary variable to store each of the items in taskInfo, so that you can give a more descriptive label to each item.
```
   String description = taskInfo[2];
   String deadline =  taskInfo[3];
   convertedTask = new Deadline(taskInfo[2], taskInfo[3]);
```

Also, there is fall-through else statement that doesn't describe the logic. From a  glance, the reviewer would not know what the value of taskInfo[0] would be if it reaches the last else statement (other than it is not "T" or "E").

Lastly, I think there's a bug here. If you input E but supply insufficient arguments, your app might crash.